src/lib/controller/index.test.ts

```diff
@@ -58,3 +61,64 @@ describe("Controller", () => {
     expect(mockHooks.onReady).toHaveBeenCalled();
   });
 });
+
+describe("getContextAndLog", () => {
```

Comment:
This batch of 5 test cases may be a good opportunity to use a table-driven testing pattern with `it.each()`. I think the "act" step in all of them is the same and we're just adjusting inputs to cover several different cases.

This change could be worth it to achieve a little more brevity while providing the same level of confidence.